packages/components/drawer/__tests__/drawer.test.tsx

```diff
@@ -0,0 +1,356 @@
+    expect(footerBtns[1].find('span').element.textContent).toEqual('confirm')
+  })
+
+  test('should append to body, when append-to-body flag is true', async () => {
+    const wrapper = mount({
+      data: () => ({ visible: false }),
+      render() {
+        return (
+          <Drawer
+            title={title}
+            modelValue={this.visible}
+            append-to-body={true}
+          />
+        )
+      },
+    })
+
+    const vm = wrapper.vm as any
+
+    vm.visible = true
+    await nextTick()
+    await rAF()
+    await nextTick()
+    expect(document.querySelector('.el-overlay')?.parentNode).toEqual(
+      document.body
+    )
+  })
+
+  test('should open and close drawer properly', async () => {
+    const onClose = vi.fn()
+    const onClosed = vi.fn()
+    const onOpened = vi.fn()
+
+    const wrapper = mount({
+      data: () => ({ visible: false }),
+      render() {
+        return (
+          <Drawer
+            title={title}
+            modelValue={this.visible}
+            onClose={onClose}
+            onClosed={onClosed}
+            onOpened={onOpened}
+          />
+        )
+      },
+    })
+
+    const vm = wrapper.vm as any
+    await nextTick()
+    await rAF()
+    await nextTick()
+    expect(onOpened).not.toHaveBeenCalled()
+
+    const drawerEl = wrapper.find('.el-overlay').element as HTMLDivElement
+    expect(drawerEl.style.display).toEqual('none')
+
+    vm.visible = true
+    await nextTick()
+    await rAF()
+    expect(drawerEl.style.display).not.toEqual('none')
+    expect(onOpened).toHaveBeenCalled()
+  })
+
+  test('should destroy every child after drawer was closed when destroy-on-close flag is true', async () => {
+    const wrapper = mount(() => (
+      <Drawer
+        ref="drawer"
+        title={title}
+        modelValue={true}
+        append-to-body={false}
+        destroy-on-close={true}
+      >
+        <span>{content}</span>
+      </Drawer>
+    ))
+    await nextTick()
+    await rAF()
+    await nextTick()
+    expect(wrapper.find('.el-drawer__body span').element.textContent).toEqual(
+      content
+    )
+
+    await wrapper.find('.el-drawer__close-btn').trigger('click')
+    await nextTick()
+    await rAF()
+    await nextTick()
+    expect(wrapper.find('.el-drawer__body').exists()).toBe(false)
+  })
+
+  test('should close dialog by clicking the close button', async () => {
+    const wrapper = mount({
+      data: () => ({ visible: true }),
+      render() {
+        return (
+          <Drawer
+            title={title}
+            modelValue={this.visible}
+            onUpdate:modelValue={(val) => (this.visible = val)}
+          />
+        )
+      },
+    })
+    await nextTick()
+    await rAF()
+    await nextTick()
+    const vm = wrapper.vm as any
+
+    await wrapper.find('.el-drawer__close-btn').trigger('click')
+    await nextTick()
+    await rAF()
+    await nextTick()
+    expect(vm.visible).toEqual(false)
+  })
```

Comment:
1. Missing `destroy-on-close=true`
2. Prefer using ref instead of data property
3. use `v-model` instead of `modelValue` and `onUpdate:modelValue`

```suggestion
  test('should close dialog by clicking the close button', async () => {
    const visible = ref(true)
    const wrapper = mount(() => (
      <Drawer title={title} v-model={visible.value} destroy-on-close />
    ))

    await nextTick()
    await rAF()
    await nextTick()

    await wrapper.find('.el-drawer__close-btn').trigger('click')
    await nextTick()
    await rAF()
    await nextTick()
    expect(visible.value).toEqual(false)
  })
```